### ISSUE TITLE
[Linux] Retroarch doesn't respect monitor index.

### ISSUE THREAD

**Issue by kenshen112** [libretro/RetroArch#6271]
I have a weak dell laptop with the intel hd 4 series gpu and a small tv as a second monitor running fedora with retroarch installed via. flatpak I have the monitor index set correctly in retroarch but if I'm using plasma 5 or i3 it renders to the smaller screen regardless although I can move it in i3, xfce it'll render to the "Main" screen the tv but if I adjust fullscreen modes like windowed  fullscreen etc. it moves back to small screen.

Adjusting Monitor index moves retroarch to larger display

Retroarch flicker's like it's attempting to do something then it stays on the small screen regardless. 

install via flatpak with intel gpu on fedora 
attempt to adjust monitor index
nothing.

fedora workstation 27 x86-64

**Comment by noooq** [comment]
Same thing happens here, Debian 9.

**Comment by kenshen112** [comment]
@noooq using flatpak as well?

**Comment by noooq** [comment]
@kenshen112 No. The install mode is irrelevant. I have seen this in the outdated debian repo version and in a version compiled from source (1.7.0)

**Comment by kenshen112** [comment]
@noooq K thanks I wasn't sure if it was somthing with flatpak's sandboxing or somthing well dev's the issues open :-)

**Comment by casdevel** [comment]
Some additional info (maybe someone find it useful):

Desktop computer, Kubuntu 16.04.03 64-bit, RetroArch v1.7.0 (Git version 2b4b38a), 'gl' video driver, AMD graphics card, X.Org driver, primary display on VGA port, secondary display (TV) on HDMI port.

'Monitor Index' option work as expected, regardless of 'Windowed Fullscreen Mode' option.

**Comment by domsleee** [comment]
I'm seeing similar problems when installing from the snap store - when in fullscreen, changing the monitor index does not move it to the correct display

I'm running Fedora 32 w/ XOrg on a dual monitor setup (amd GPU)

I found that changing the flag "Windowed Mode" --> "Show Window Decorations" moved the program between my two monitors. Monitor index had no effect (both when show window decorations was on and off)

**Comment by lucasmr** [comment]
I'm having the exact same behavior as @domsleee on RetroArch 1.9.0.

**Comment by KingDuckZ** [comment]
I have the same issue. I wanted to keep retroarch on the TV screen and on IRC they suggested looking into the monitor index setting but all numbers I tried do nothing at all.
I'm running retroarch built manually from source on current master. OS is Gentoo with mesa Radeon drivers.

```
./configure --enable-systemd --enable-dbus --enable-freetype --enable-pulse --enable-sse --enable-vulkan --enable-threads --enable-spirv_cross --enable-qt --enable-flac --enable-v4l2 --enable-al --enable-zlib --enable-alsa --enable-x11 --enable-sdl2
make -j8
```

**Comment by Sixdd6** [comment]
Still seeing this issue with 1.9.0 from the Snap store on Manjaro, linux kernel 5.10.23, nvidia rtx 2070 super running driver 460.56. Reguardless of which monitor is primary retroarch still only displays on the left-most monitor for me. Note however that when I installed through the official repo with pacman it actually did change monitors correctly but since that version doesn't include the online core downloader it's a pain to use.

**Comment by gouchi** [comment]
The difference I see between the Arch and Snap packages is that we did not enable `libxinerama` and/or `libxrandr` in the Snap package.

So it will worth to try with those libraries enabled.

**Comment by KingDuckZ** [comment]
I've been building it myself, since I'm also annoyed by the fact packaged version for Gentoo is missing half the cores and is very outdated. I just rebase from master periodically and build that. So is there a specific configure flag I need to pass in order for this to work?

**Comment by gouchi** [comment]
Hi,

They are autodetected

```
Checking presence of package xinerama ... 1.1.4
Checking existence of -lXrandr ... yes
```
but you can enable them with ` --enable-xinerama --enable-xrandr` if needed.

**Comment by KingDuckZ** [comment]
It turns out these two were already enabled, this is how I've been compiling so far:

```
./configure --enable-systemd --enable-dbus --enable-freetype --enable-pulse --enable-sse --enable-vulkan --enable-threads --enable-spirv_cross --enable-qt --enable-flac --enable-v4l2 --enable-al --enable-zlib --enable-alsa --enable-x11 --enable-sdl2 --enable-xinerama --enable-xrandr
make -j8
```

It's still defaulting to the laptop screen, and even if I move retroarch manually to the hdmi screen, it goes back each time I start or close a game.

**Comment by hallamjeff** [comment]
I also have this issue on Ubuntu MATE 21.04 with RetroArch v1.9.4 installed as a snap package. Intel Xe graphics.

**Comment by freeerasure** [comment]
Also have this problem on Windows 10 x64 version. This issue has been open for 4 years now and nothing done with it. Surprise surprise.

**Comment by gouchi** [comment]
On Xorg, it is working using [fluxbox](http://fluxbox.org/) but not in [i3wm](https://i3wm.org/) whatever the settings fullscreen on/off, windowed fullscreen on/off, windows decoration on/off.

**Comment by 2V3EvG4LMJFdRe** [comment]
On macOS 10.13.6 I have to use Mission Control to do this. `video_monitor_index =` values 1 and above open ra on the first monitor with a permanent blank screen and the program requires force quit to fully exit it. No value opens ra on the second monitor.

**Comment by gouchi** [comment]
@2V3EvG4LMJFdRe This is issue is related to Linux not macOS. So it would be better to open a dedicated issue for macOS.

Thank you.

**Comment by danmcadams** [comment]
2025 and still not fixed. Ugh.

**Comment by lousuan** [comment]
Same problem here... Im using steam version of retroarch on Ubuntu 24.10.

**Comment by gouchi** [comment]
@danmcadams @lousuan Please provide some information at least the display server and window manager you are using otherwise it brings nothing to say it does not work.

Thank you for your cooperation.

**Comment by lousuan** [comment]
> [@danmcadams](https://github.com/danmcadams) [@lousuan](https://github.com/lousuan) Please provide some information at least the display server and window manager you are using otherwise it brings nothing to say it does not work.
> 
> Thank you for your cooperation.

Im using gnome and x11. What information do you need? or any command you want me to type in bash?

**Comment by HereInPlainSight** [comment]
Gentoo user, having... slightly contrary results from the other Gentoo user, but maybe something's changed.  Ebuild shouldn't be _too_ different, but mine's from GURU and not self-made.  All tests were using the `vulkan` video driver, I did not test with `gl`.

I was having issues with the following use flags, according to `eix games-emulation/RetroArch`:
`7zip X alsa cheevos egl ffmpeg materialui network opengl ozone pipewire rgui sdl threads truetype udev vulkan wayland xv zlib -cg -debug -dispmanx -filters -gles2 -gles3 -hid -jack -kms -libass -libusb -openal -oss -pulseaudio -v4l2 -videocore -xinerama -xmb`

I tried the patch from the other thread since it was more recent than the comments here, but no dice.  So I rolled that back and went back through recommendations in this thread, and after adding only xinerama to my existing use flags, it started working.

Current use flags from `eix`:
`7zip X alsa cheevos egl ffmpeg materialui network opengl ozone pipewire rgui sdl threads truetype udev vulkan wayland xinerama xv zlib -cg -debug -dispmanx -filters -gles2 -gles3 -hid -jack -kms -libass -libusb -openal -oss -pulseaudio -v4l2 -videocore -xmb`

The config statement from the ebuild running:
```
>>> Configuring source in /var/tmp/portage/games-emulation/RetroArch-1.21.0/work/RetroArch-1.21.0 ...
 * ./configure --prefix=/usr --docdir=/usr/share/doc/RetroArch-1.21.0 --enable-dynamic --disable-builtinzlib --disable-qt --disable-sdl --disable-vg --disable-osmesa --enable-7zip --enable-alsa --enable-cheevos --disable-cg --disable-neon --disable-floathard --enable-sse --disable-dispmanx --enable-egl --enable-ffmpeg --disable-opengles --disable-opengles3 --disable-hid --disable-jack --disable-kms --disable-ssa --disable-libusb --enable-materialui --enable-networking --disable-al --enable-opengl --disable-oss --enable-ozone --disable-pulse --enable-pipewire --enable-rgui --enable-sdl2 --enable-threads --enable-freetype --enable-udev --disable-v4l2 --disable-videocore --enable-vulkan --enable-wayland --enable-x11 --enable-xinerama --disable-xmb --enable-xvideo --enable-zlib
```

As an odd addendum...  It _works_, but it's a little weird.  I have an `ASUS GeForce RTX 4070 SUPER`, which has four outputs.  The first three options (0, 1, and  2) all select the correct monitor now (my default in KDE Plasma), and anything 3+ chooses my secondary monitor.  (I don't know how high it goes -- I got to 25 and stopped.  <.<  Internet seems to imply this is normally capped at a low number, reddit searches indicated people had to manually set the config file to '2' sometimes because it wouldn't go that high naturally, not sure how the determination of the max is made or if it's Just Linux Things™.)

Anyway, just wanted to add some data to the pool.
